src/mudlet.cpp

```diff
@@ -2378,19 +2369,20 @@ QString mudlet::readProfileData( const QString& profile, const QString& item )
 // this slot is called via a timer in the constructor of mudlet::mudlet()
 void mudlet::startAutoLogin()
 {
-    QStringList hostList = QDir(QDir::homePath()+"/.config/mudlet/profiles").entryList(QDir::Dirs, QDir::Name);
-    hostList.removeAll(".");
-    hostList.removeAll("..");
-    for( int i = 0; i< hostList.size(); i++ )
-    {
-        QString item = "autologin";
-        QString val = readProfileData( hostList[i], item );
-        if( val.toInt() == Qt::Checked )
-        {
-            doAutoLogin( hostList[i] );
+    QStringList hostList = QDir(QDir::homePath() + "/.config/mudlet/profiles").entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+    bool openedProfile = false;
+
+    for (auto host : hostList) {
+        QString val = readProfileData(host, QStringLiteral("autologin"));
+        if (val.toInt() == Qt::Checked) {
```

Comment:
Hang on - if that item is not present (has not ever been set) then I think we get back a null `QString()` for `val`.  Using a `toInt()` on _that_ will fail and return `0` which just happens to be the same as the `Qt::Unchecked` `enum` {`Qt::Checked` is `2` currently in comparison} - that being the case, should we not be a _little_ more rigorous and use:

    if ((! val.isEmpty()) && val.toInt() == Qt::Checked) {

?